cpp/examples/advanced/cugraph_operations/graph_operations.cu

```diff
@@ -0,0 +1,253 @@
+
+    rmm::device_uvector<result_t> outputs(size_of_the_vertex_partition_assigned_to_this_process,
+                                          handle.get_stream());
+
+    per_v_transform_reduce_incoming_e(
+      handle,
+      graph_view,
+      src_vertex_weights_cache.view(),
+      dst_vertex_weights_cache.view(),
+      (*edge_weight_view),
+      [] __device__(auto src, auto dst, auto src_prop, auto dst_prop, auto edge_prop) {
+        printf("\nsrc ---> %d dst = %d :  src_prop = %f dst_prop = %f edge_prop = %f\n",
+               static_cast<int>(src),
+               static_cast<int>(dst),
+               static_cast<float>(src_prop),
+               static_cast<float>(dst_prop),
+               static_cast<float>(edge_prop));
+        return dst_prop * edge_prop;
+      },
+      result_t{0},
+      cugraph::reduce_op::plus<result_t>{},
+      outputs.begin());
+
+    auto outputs_title                 = std::string("outputs_").append(std::to_string(comm_rank));
+    size_t max_nr_of_elements_to_print = 10;
+    RAFT_CUDA_TRY(cudaDeviceSynchronize());
+    raft::print_device_vector(outputs_title.c_str(),
+                              outputs.begin(),
+                              std::min<size_t>(outputs.size(), max_nr_of_elements_to_print),
+                              std::cout);
+  } else {
+    using result_t      = edge_t;
+    auto vertex_weights = graph_view.compute_out_degrees(handle);
+
+    cugraph::edge_src_property_t<graph_view_t, result_t> src_vertex_weights_cache(handle,
+                                                                                  graph_view);
+    cugraph::edge_dst_property_t<graph_view_t, result_t> dst_vertex_weights_cache(handle,
+                                                                                  graph_view);
+
+    update_edge_src_property(handle, graph_view, vertex_weights.begin(), src_vertex_weights_cache);
+
+    update_edge_dst_property(handle, graph_view, vertex_weights.begin(), dst_vertex_weights_cache);
+
+    rmm::device_uvector<result_t> outputs(size_of_the_vertex_partition_assigned_to_this_process,
+                                          handle.get_stream());
+
+    per_v_transform_reduce_incoming_e(
+      handle,
+      graph_view,
+      src_vertex_weights_cache.view(),
+      dst_vertex_weights_cache.view(),
+      cugraph::edge_dummy_property_t{}.view(),
+      [] __device__(auto src, auto dst, auto src_prop, auto dst_prop, auto) {
+        printf("\nsrc ---> %d dst = %d :  src_prop = %f dst_prop = %f\n",
+               static_cast<int>(src),
+               static_cast<int>(dst),
+               static_cast<float>(src_prop),
+               static_cast<float>(dst_prop));
+        return dst_prop;
+      },
+      result_t{0},
+      cugraph::reduce_op::plus<result_t>{},
+      outputs.begin());
+
+    auto outputs_title                 = std::string("outputs_").append(std::to_string(comm_rank));
+    size_t max_nr_of_elements_to_print = 10;
```

Comment:
And considering that we are printing for every edge in the above device functor, should we really limit the elements to print here?

Reply:
Removed. Not needed anymore.